src/docs/stan-reference/programming.tex

```diff
@@ -10635,10 +10633,176 @@ \subsection{Examples: Fits in Stan}
 
 
 
-\chapter{Optimizing \Stan Code}\label{optimization.chapter}
+\chapter{Optimizing \Stan Code for Efficiency}\label{optimization.chapter}
 \noindent
 This chapter provides a grab bag of techniques for optimizing \Stan
 code, including vectorization, sufficient statistics, and conjugacy.
+At a coarse level, efficiency involves both the amount of time
+required for a computation and the amount of memory required.  For
+practical applied statistical modeling, we are mainly concerned with
+reducing wall time (how long a program takes as measured by a clock on
+the wall) and keeping memory requirements within available bounds.
+
+\section{What is Efficiency?}
+
+The standard algorithm analyses in computer science measure efficiency
+asymptotically as a function of problem size (such as data, number of
+parameters, etc.) and typically does not consider constant factors.
```

Comment:
@bob-carpenter I think "does not" should be "do not" (if using "analyses" instead of "analysis" above